src/docs/release/breaking-changes/use-maxLengthEnforcement-instead-of-maxLengthEnforced.md

```diff
@@ -0,0 +1,208 @@
+characters counter.
+
+Although `maxLengthEnforced` allows users to control the behavior of the
+truncation, so does the `LengthLimitingTextInputFormatter`, they didn't handle
+the truncate behavior for CJK (Chinese, Japanese, and Korean) characters
+properly, while these characters require users to type a sequence of Latin
+characters to enter one such character, which will cause composing CJK
+characters failed to produce proposing characters.
+
+In order to solve the breaking behavior about CJK's truncation, a new tri-state
+enum `MaxLengthEnforcement` has introduced, which describes supported strategies
+for handling active composing region when applying a
+`LengthLimitingTextInutFormatter`. This enum has been added to text fields to
+replace the `maxLengthEnforced` parameter. With the new enum parameter,
+developers can choose different strategies based on the type of the content
+the text field expects.
+
+(For more information, see the docs for [`maxLength`][] and
+[`MaxLengthEnforcement`][].)
+
+The default max length truncation behavior is based on the `TargetPlatform`
+that you're using or running on.
+
+## Description of change
+
+* Added a `MaxLengthEnforcement` enum as a replacement for the now-deprecated
+  boolean `maxLengthEnforced` parameter on `TextField`, `TextFormField`,
+  and `CupertinoTextField`.
+* The `maxLengthEnforced` property in editable text field classes is deprecated.
+* A named optional parameter, `maxLengthEnforcement` is added to
+  the `TextField`, `TextFormField`, `CupertinoTextField`,
+  and `LengthLimitingTextInputFormatter` classes.
+* The default value of the maxLengthEnforced property is inferred based on the
+  target platform, to improve the input experience:
+  * Android, Windows: defaults to `MaxLengthEnforcement.enforced`.
+  * iOS, Web, MacOS, Linux, Fuchsia: defaults to
+    `MaxLengthEnforcement.truncateAfterCompositionEnds`.
+
+## Migration guide
+
+The following tips describe how to produce common behaviors, and which behavior
+is using on different platforms by default. Notice that the enforcement API is
+exposed to `TextField`, `TextFormField`, `CupertinoTextField` and
+`LengthLimitingTextInputFormatter`.
+
+_Use the default behavior for the current platform is recommended, since it
+won't break someone that is not surposed to input like the behavior that
+developers has defined._
+
+### Platforms behavior
+
+* Android, Windows: `MaxLengthEnforcement.enforced`. The native behavior of
+  these platforms is enforced. The composing will be handled by the IME while
+  users are entering CJK characters.
+* iOS: `MaxLengthEnforcement.truncateAfterCompositionEnds`. iOS has no default
+  behavior and it requires users implement the behavior themselves. Allow the
+  composition to exceed to avoid breaking CJK input.
+* Web, MacOS, Linux, Fuchsia: `MaxLengthEnforcement.truncateAfterCompositionEnds`.
+  These platforms allow the composition to exceed by default.
```

Comment:
I'm a bit curious about macOS and Fuchsia. I'd expect macOS text fields to not have a `maxLength` attribute so it falls into the same category as iOS (but I'm not sure).

Reply:
* MacOS: Yes it's the same as iOS because maybe they both require developers handle it manually.
* Fuchsia: Because it's flutter native, I simply make the definition to make it as default to avoid break CJKs.